Source/Plugins/Core/com.equella.core/js/tsrc/modules/SelectionSessionModule.ts

```diff
@@ -15,26 +15,48 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { AppConfig, SelectionSessionInfo } from "../AppConfig";
+import { AppConfig, getRenderData, SelectionSessionInfo } from "../AppConfig";
+
+const selectionSessionInfo:
+  | SelectionSessionInfo
+  | undefined
+  | null = getRenderData()?.selectionSessionInfo;
+
+/**
+ * Type guard to check whether an object is of type SelectionSessionInfo.
+ * @param data The data to be checked
+ */
+const isSelectionSession = (
+  data: unknown
+): data is { [K in keyof SelectionSessionInfo]: unknown } =>
+  typeof data === "object" && data !== null && "stateId" in data;
+
+/**
+ * True if the Selection Session info provided by renderData is neither null nor undefined.
+ */
+export const inSelectionSession = isSelectionSession(selectionSessionInfo);
```

Comment:
Be good to force the issue here - also for the sake of doco:

```suggestion
export const inSelectionSession: boolean = isSelectionSession(selectionSessionInfo);
```

Reply:
I have to change this const to a function to make tests passed. As a const, it gets evaluated during those imports (SearchResult imports this from the module) and the mocks done in the test are too late. So making it a function can help solve this issue.

Reply:
Ha! I almost commented on this during the review, but then decided the single evaluation would be fine. I forgot about the test side of things. ;)